File: Project3-Java/PageTable.java
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class PageTable {
    List<NodePageTable> table;
    int sizeProcess;
    int sizeOfPage;

                // &&&&&&&&&&&&&&&&&&&& Constructor
                // &&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&
    public PageTable() {
        Config config = new Config();
        this.sizeProcess = config.getSizeProcess();
        this.sizeOfPage = config.getSizeOfPage();
        table = new ArrayList<>(sizeProcess/sizeOfPage);
        for (int i =0 ; i<sizeProcess/sizeOfPage;i++)
            table.add(new NodePageTable(0,0));
    }
    public void add(int numberPage,int numberFrame){
        table.set(numberPage,new NodePageTable(1,numberFrame));
    }
    public void remove(int numberPage){
        table.get(numberPage).hasValue = 0;
    }
    public int findFrame(int numberPage){
        if (table.get(numberPage).hasValue == 1)
            return table.get(numberPage).getNumberFrame();
        else
            return -1;
    }
                // ******************************************
                // **************** Getter & Setter
                // ******************************************

    public List<NodePageTable> getTable() {
        return table;
    }



    public int getSizeProcess() {
        return sizeProcess;
    }



    public int getSizeOfPage() {
        return sizeOfPage;
    }


}

File: Project3-Python/PageTable.py
from Config import Config


class NodePageTable:
    hasValue = None;
    numberFrame = None;

    def __init__(self,hasValue,numberFrame) -> None:
        self.hasValue = hasValue;
        self.numberFrame = numberFrame;


class PageTable:
    sizeProcess = None
    sizePage = None
    table = list()
    def __init__(self) -> None:
        config = Config();
        self.sizePage = config.sizePage;
        self.sizeProcess = config.sizeProcess;
        for i in range(self.sizeProcess // self.sizePage):
            self.table.append(NodePageTable(0,0))

    def add(self,numberPage,numberFrame):
        self.table[numberPage]= NodePageTable(1,numberFrame);


    def remove(self,numberPage):
        self.table[numberPage].hasValue = 0

    def findFrame(self, numberPage):
        if self.table[numberPage].hasValue == 1:
            return self.table[numberPage].numberFrame;
        else:
            return -1;
